src/components/Timer/Timer.tsx

```diff
@@ -0,0 +1,36 @@
+import React from "react";
+import { TimerWrapper, TimerH3, TimerH1, TimerRectangle } from "./Timer.style";
+import rectangle1 from "./assets/Rectangle.svg";
+import rectangle2 from "./assets/Rectangle2.svg";
+
+import { useState, useEffect } from "react";
+
+export interface ITimer {
+  player: string;
+  time: number;
+}
+
+const Timer: React.FC<ITimer> = (props) => {
+  const { player } = props;
+
+  const [timeLeft, setTimeLeft] = useState(30);
+
+  useEffect(() => {
+
+    if (timeLeft === 0){ return }
+    const interval = setInterval(() => {
+      setTimeLeft(() => timeLeft - 1);
```

Comment:
This should be
```js
setTimeLeft(timeLeft => timeLeft - 1);
```
This way the setInterval would update the state with the proper value, and you won't need `timeLeft` in the dependency array anymore